extractor/filesystem/language/swift/swiftutils/podfilelock.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package swiftutils
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// PodfileLock represents the structure of a Podfile.lock file.
+type PodfileLock struct {
+	Pods            []interface{}     `yaml:"PODS"`
+	SpecChecksums   map[string]string `yaml:"SPEC CHECKSUMS"`
+	PodfileChecksum string            `yaml:"PODFILE CHECKSUM"`
+	Cocopods        string            `yaml:"COCOAPODS"`
+}
+
+// Package represents a single package parsed from Podfile.lock.
+type Package struct {
+	Name    string
+	Version string
+}
+
+// ParsePodfileLock parses the contents of a Podfile.lock and returns a list of packages.
+func ParsePodfileLock(reader io.Reader) ([]Package, error) {
+	bytes, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read file: %w", err)
+	}
+
+	// Check if the file is empty
+	if len(bytes) == 0 {
+		return nil, fmt.Errorf("file is empty")
+	}
+
+	var podfile PodfileLock
+	if err = yaml.Unmarshal(bytes, &podfile); err != nil {
+		return nil, fmt.Errorf("unable to parse YAML: %w", err)
+	}
+
+	var pkgs []Package
+	for _, podInterface := range podfile.Pods {
+		var podBlob string
+		switch v := podInterface.(type) {
+		case map[string]interface{}:
+			for k := range v {
+				podBlob = k
+			}
+		case string:
+			podBlob = v
```

Comment:
In the example valid file we always have the map[string] case right? Can you add an example that has the string case?